numpy/testing/utils.py

```diff
@@ -395,13 +414,25 @@ def assert_equal(actual,desired,err_msg='',verbose=True):
     except (TypeError, ValueError, NotImplementedError):
         pass
 
-    # Explicitly use __eq__ for comparison, ticket #2552
-    with suppress_warnings() as sup:
-        # TODO: Better handling will to needed when change happens!
-        sup.filter(DeprecationWarning, ".*NAT ==")
-        sup.filter(FutureWarning, ".*NAT ==")
-        if not (desired == actual):
-            raise AssertionError(msg)
+    # Put lock around the warning filter, see comment at lock definition
+    if _assert_comparison_lock is None:
+        # Explicitly use __eq__ for comparison, ticket #2552
+        with suppress_warnings() as sup:
+            # TODO: Better handling will to needed when change happens!
+            sup.filter(DeprecationWarning, ".*NAT ==")
+            sup.filter(FutureWarning, ".*NAT ==")
+            if not (desired == actual):
+                raise AssertionError(msg)
+        return
+    with _assert_comparison_lock:
```

Comment:
I suppose the lock fix could also be done in skimage, its a bit tricky. This only band-aids the cases in skimage, might band-aid some other cases, but it is definitely not a real fix....